common/helpers.go

```diff
@@ -142,26 +142,25 @@ func createWaitForEventHandler(
 	ch := make(chan interface{})
 
 	go func() {
-		for {
-			select {
-			case <-evCancelCtx.Done():
-				return
-			case ev := <-chEvHandler:
-				if stringSliceContains(events, ev.typ) {
-					if predicateFn != nil {
-						if predicateFn(ev.data) {
-							ch <- ev.data
-						}
-					} else {
-						ch <- nil
```

Comment:
Any reason for removing this loop here? This function may receive other events as well (besides the ones `predicateFn` is looking for)? If so, might it return if it finds an irrelevant event first and misses the target event that the `predicateFn` is looking for?

Reply:
Hhmm you're right, nice catch :+1: My initial thought was that it returned in both cases, so the loop wasn't needed, but on closer look it definitely should keep running :facepalm:

Thanks for looking into it!

Reply:
Sure thing, glad I could help :)